Source/Scene/PointPrimitiveCollection.js

```diff
@@ -0,0 +1,976 @@
+        this._pointPrimitivesToUpdateIndex = 0;
+        this._pointPrimitivesRemoved = false;
+        this._createVertexArray = false;
+
+        this._shaderScaleByDistance = false;
+        this._compiledShaderScaleByDistance = false;
+        this._compiledShaderScaleByDistancePick = false;
+
+        this._shaderTranslucencyByDistance = false;
+        this._compiledShaderTranslucencyByDistance = false;
+        this._compiledShaderTranslucencyByDistancePick = false;
+
+        this._propertiesChanged = new Uint32Array(NUMBER_OF_PROPERTIES);
+
+        this._maxPixelSize = 1.0;
+
+        this._baseVolume = new BoundingSphere();
+        this._baseVolumeWC = new BoundingSphere();
+        this._baseVolume2D = new BoundingSphere();
+        this._boundingVolume = new BoundingSphere();
+        this._boundingVolumeDirty = false;
+
+        this._colorCommands = [];
+        this._pickCommands = [];
+
+        /**
+         * The 4x4 transformation matrix that transforms each pointPrimitive in this collection from model to world coordinates.
+         * When this is the identity matrix, the pointPrimitives are drawn in world coordinates, i.e., Earth's WGS84 coordinates.
+         * Local reference frames can be used by providing a different transformation matrix, like that returned
+         * by {@link Transforms.eastNorthUpToFixedFrame}.
+         *
+         * @type {Matrix4}
+         * @default {@link Matrix4.IDENTITY}
+         *
+         * @see Transforms.eastNorthUpToFixedFrame
+         *
+         * @example
+         * var center = Cesium.Cartesian3.fromDegrees(-75.59777, 40.03883);
+         * pointPrimitives.modelMatrix = Cesium.Transforms.eastNorthUpToFixedFrame(center);
+         * pointPrimitives.add({
+         *   color : Cesium.Color.ORANGE,
+         *   position : new Cesium.Cartesian3(0.0, 0.0, 0.0) // center
+         * });
+         * pointPrimitives.add({
+         *   color : Cesium.Color.YELLOW,
+         *   position : new Cesium.Cartesian3(1000000.0, 0.0, 0.0) // east
+         * });
+         * pointPrimitives.add({
+         *   color : Cesium.Color.GREEN,
+         *   position : new Cesium.Cartesian3(0.0, 1000000.0, 0.0) // north
+         * });
+         * pointPrimitives.add({
+         *   color : Cesium.Color.CYAN,
+         *   position : new Cesium.Cartesian3(0.0, 0.0, 1000000.0) // up
+         * });
+         */
+        this.modelMatrix = Matrix4.clone(defaultValue(options.modelMatrix, Matrix4.IDENTITY));
+        this._modelMatrix = Matrix4.clone(Matrix4.IDENTITY);
+
+        /**
+         * This property is for debugging only; it is not for production use nor is it optimized.
+         * <p>
+         * Draws the bounding sphere for each draw command in the primitive.
+         * </p>
+         *
+         * @type {Boolean}
+         *
+         * @default false
+         */
+        this.debugShowBoundingVolume = defaultValue(options.debugShowBoundingVolume, false);
```

Comment:
It looks... possibly wrong to me, but it's hard to tell since the sphere itself is distorted by the CV projection.  The placement of the sphere is covered by tests, though.